mergify_engine/queue.py

```diff
@@ -37,42 +37,34 @@ class QueueConfig(typing.TypedDict):
 
 @dataclasses.dataclass
 class Queue:
-    redis: utils.RedisCache
-    owner_id: int
-    owner: str
-    repo_id: int
-    repo: str
+    repository: context.Repository
     ref: str
 
     log: logging.LoggerAdapter = dataclasses.field(init=False)
 
     def __post_init__(self):
         self.log = daiquiri.getLogger(
-            __name__, gh_owner=self.owner, gh_repo=self.repo, gh_branch=self.ref
+            __name__,
+            gh_owner=self.repository.installation.owner_login,
+            gh_repo=self.repository.name,
+            gh_branch=self.ref,
         )
 
     @classmethod
     def from_context(cls, ctxt: context.Context) -> "Queue":
-        return cls(
-            ctxt.redis,
-            ctxt.pull["base"]["repo"]["owner"]["id"],
-            ctxt.pull["base"]["repo"]["owner"]["login"],
-            ctxt.pull["base"]["repo"]["id"],
-            ctxt.pull["base"]["repo"]["name"],
-            ctxt.pull["base"]["ref"],
-        )
+        return cls(ctxt.repository, ctxt.pull["base"]["ref"])
```

Comment:
I'd use a `Repository` as arg here, it looks overkill to require an entire `Context`.

Reply:
The method is called `from_context` :p

Reply:
I guess you could rename it?